### DISCUSSION TITLE
Can It blocks be used in BeforeAll/BeforeEach etc?

### DISCUSSION THREAD

**Discussion by codykonior** [pester/Pester#2132] [Q&A] [answered]
Say you want to have an It test that your module loads. But you're also loading the module in the BeforeAll block. Can you wrap it in an It there?

I could not find a definitive list of what blocks are allowed to be nested inside each other.

**Comment by nohwnd** [comment, marked as answer]
You should not, it won't find the test, technically it is possible because tests run after BeforeAll. But having at least one test in the respective block is a pre-condition to run a BeforeAll at all. So even though I can make it work easily it by changing the internal test tree from BeforeAll, I did not pursue this idea further. So it is not supported and won't find your test.

```powershell
Invoke-Pester -Container (New-PesterContainer -ScriptBlock {
    Describe "a" { 
        BeforeAll {
            It "a" {}
        }
        It "b" {}
    }
}) -Output Detailed
```
```
Pester v5.3.0

Starting discovery in 1 files.
Discovery found 1 tests in 6ms.
Running tests.
Describing a
  [+] b 4ms (1ms|3ms)
Tests completed in 31ms
Tests Passed: 1, Failed: 0, Skipped: 0 NotRun: 1
```
